File: src/2025-03-04-03-28-40--deepseek-chat-0304/cpp/exercises/practice/kindergarten-garden/kindergarten_garden.cpp
#include "kindergarten_garden.h"
#include <algorithm>
#include <map>

namespace kindergarten_garden {

const std::vector<std::string> children = {
    "Alice", "Bob", "Charlie", "David",
    "Eve", "Fred", "Ginny", "Harriet",
    "Ileana", "Joseph", "Kincaid", "Larry"
};

const std::map<char, Plants> plant_map = {
    {'G', Plants::grass},
    {'C', Plants::clover},
    {'R', Plants::radishes},
    {'V', Plants::violets}
};

std::array<Plants, 4> plants(const std::string& diagram, const std::string& child) {
    std::string row1, row2;
    size_t newline = diagram.find('\n');
    row1 = diagram.substr(0, newline);
    row2 = diagram.substr(newline + 1);
    
    // Find child's index
    auto it = std::find(children.begin(), children.end(), child);
    if (it == children.end()) {
        return {};
    }
    size_t index = std::distance(children.begin(), it);
    size_t pos = index * 2;
    
    if (pos + 1 >= row1.length()) {
        return {};
    }
    
    return {
        plant_map.at(row1[pos]),
        plant_map.at(row1[pos + 1]),
        plant_map.at(row2[pos]),
        plant_map.at(row2[pos + 1])
    };
}

}  // namespace kindergarten_garden

File: src/2025-03-04-03-28-40--deepseek-chat-0304/java/exercises/practice/kindergarten-garden/src/main/java/KindergartenGarden.java
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Map;
import java.util.HashMap;

class KindergartenGarden {
    
    private static final String[] STUDENTS = {
        "Alice", "Bob", "Charlie", "David",
        "Eve", "Fred", "Ginny", "Harriet",
        "Ileana", "Joseph", "Kincaid", "Larry"
    };
    
    private static final Map<Character, Plant> PLANT_MAP = Map.of(
        'G', Plant.GRASS,
        'C', Plant.CLOVER,
        'R', Plant.RADISHES,
        'V', Plant.VIOLETS
    );
    
    private Map<String, List<Plant>> studentPlants;

    KindergartenGarden(String garden) {
        this(garden, STUDENTS);
    }

    KindergartenGarden(String garden, String[] students) {
        if (garden == null || students == null) {
            throw new IllegalArgumentException("Garden and students cannot be null");
        }

        studentPlants = new HashMap<>();
        String[] rows = garden.split("\n");
        
        // Ensure we have exactly two rows
        if (rows.length != 2) {
            throw new IllegalArgumentException("Garden must have exactly two rows");
        }

        // Sort students alphabetically
        Arrays.sort(students);
        
        for (int i = 0; i < students.length; i++) {
            List<Plant> plants = new ArrayList<>();
            
            // Calculate positions
            int pos1 = i * 2;
            int pos2 = i * 2 + 1;
            
            // Add plants from first row if positions are valid
            if (pos1 < rows[0].length() && pos2 < rows[0].length()) {
                plants.add(PLANT_MAP.getOrDefault(rows[0].charAt(pos1), Plant.GRASS));
                plants.add(PLANT_MAP.getOrDefault(rows[0].charAt(pos2), Plant.GRASS));
            }
            
            // Add plants from second row if positions are valid
            if (pos1 < rows[1].length() && pos2 < rows[1].length()) {
                plants.add(PLANT_MAP.getOrDefault(rows[1].charAt(pos1), Plant.GRASS));
                plants.add(PLANT_MAP.getOrDefault(rows[1].charAt(pos2), Plant.GRASS));
            }
            
            studentPlants.put(students[i], plants);
        }
    }

    List<Plant> getPlantsOfStudent(String student) {
        if (student == null) {
            throw new IllegalArgumentException("Student cannot be null");
        }
        return studentPlants.getOrDefault(student, List.of());
    }
}
